packages/amplify_auth_cognito/test/amplify_auth_cognito_confirmSignIn_test.dart

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_auth_cognito/method_channel_auth_cognito.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+extension IsEqual on SignInResult {
+  bool isMostlyEqual(SignInResult comparator) {
+    if(
+      comparator.isSignedIn == isSignedIn &&
+      comparator.nextStep.signInStep == nextStep.signInStep &&
+      comparator.nextStep.additionalInfo.length == nextStep.additionalInfo.length &&
+      comparator.nextStep.codeDeliveryDetails.destination == nextStep.codeDeliveryDetails.destination &&
+      comparator.nextStep.codeDeliveryDetails.attributeName == nextStep.codeDeliveryDetails.attributeName &&
+      comparator.nextStep.codeDeliveryDetails.deliveryMedium == nextStep.codeDeliveryDetails.deliveryMedium
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
+
+void main() {
+  const MethodChannel authChannel = MethodChannel('com.amazonaws.amplify/auth_cognito');
+  const MethodChannel coreChannel = MethodChannel('com.amazonaws.amplify/core');
+
+  Amplify amplify = new Amplify();
+  AmplifyAuthCognito auth = AmplifyAuthCognito();
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUp(() {
+    coreChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      return true;
+    });
+  });
+
+  tearDown(() {
+    authChannel.setMockMethodCallHandler(null);
+    coreChannel.setMockMethodCallHandler(null);
+  });
+
+  test('confirmSignIn request returns a SignInResult', () async {
+
+    authChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "confirmSignIn") {
+        assert(methodCall.arguments["data"] is Map);
+        assert(methodCall.arguments["data"]["confirmationCode"] is String);
```

Comment:
We can do more detailed assertions here to ensure that the method channel was called with right parameters (testing serialization in dart)

Reply:
These actually are the expected parameters - it's just one attribute for this method (still using a "data" map in anticipation of adding lambda metadata down the road).

Reply:
@Amplifiyer ^^

Reply:
No I mean asserting the actual values rather than the type. e.g.
`assert("iAmLegit", methodCall.arguments["data"]["confirmationCode"]); `